### PR TITLE
Release 1.0.3

### LINKED ISSUE #154: goodpractice removed from CRAN
From https://cran.r-project.org/package=goodpractice:

> Package ‘goodpractice’ was removed from the CRAN repository.
> 
> Formerly available versions can be obtained from the [archive](https://cran.r-project.org/src/contrib/Archive/goodpractice).
> 
> Archived on 2022-04-27 as check issues were not corrected in time.
> 
> A summary of the most recent check results can be obtained from the [check results archive](https://cran-archive.r-project.org/web/checks/2022/2022-04-27_check_results_goodpractice.html).

From a quick look at the check results I'd say that the error might be loosely related to #135/#136, but I might be wrong. Are you still familiar with this @russHyde?

### PR DESCRIPTION
**Pull request by owenjonesuob**
closes  #154

Still to do:

* [x] Add actions from [r-lib/actions](https://github.com/r-lib/actions)
* [x] Add tests for new code
* [x] Ensure R CMD check is coming back clean
* [x] Run through CRAN release process

---

Additions:

* Limit for cyclomatic complexity check can be adjusted using the `gp.cyclocomp.limit` option, default 50 (#132, fabian-s).
* The number of lines printed to the console by each check result can be set using the new `positions_limit` parameter into `print()` - previously it was always 5 lines (#130, fabian-s).
* GitHub Actions now used for CI/CD checks (#145).

Bugfixes:

* Documentation for custom checks significantly improved (#133, fabian-s).
* Year updated in `LICENSE`, and `LICENSE.md` added to clarify that {goodpractice} uses the MIT license (#144).

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
+             email = "daniel.debortoli@mango-solutions.com",
+             comment = c(GitHub = "ddbortoli")),
+      person(given = "Gabor",
+             family = "Csardi",
+             role = "aut",
+             email = "csardi.gabor@gmail.com"),
+      person(given = "Hannah",
+             family = "Frick",
+             role = "aut",
+             email = "hfrick@mango-solutions.com"),
```
Comment (hfrick) [inline DESCRIPTION:20]:
```suggestion
             email = "hannah.frick@gmail.com"),
```

**Comment by hfrick** [comment]
@anasimmons and @ddbortoli wanted to resubmit - maybe they can give an update here? 🙏 (some of the conversation is in the corresponding PR #151 )

**Comment by KarinaMarks** [comment]
Looking into this now with @anasimmons  - the package failed cran checks on fedora. Current issues are:

* example in docs of `failed_checks` erros
* function `results` errors sometimes in the tests (test-lintr.R, test-api.R, test-tf.R)
* running `gp` in the vignettes errors (both vignettes)

Always the same error: 
`Error in is.na(chk) || ("status" %in% names(chk) && is.na(chk[["status"]])): 'length = 2' in coercion to 'logical(1)'`

I believe this is line 72 file gp.R within function `check_passed` - will need to investigate more into why this is happening on fedora.

Any thoughts/ideas on this appreciated!

**Comment by dpprdan** [comment]
@KarinaMarks isn't this fixed already by https://github.com/MangoTheCat/goodpractice/commit/09dd1197f3a0194270f689225797e8279b412293 (which is post-v1.0.2 AFAICT)? That commit removes the 

```r
if (is.na(chk) || ("status" %in% names(chk) && is.na(chk[["status"]])))
```

line at least. See #135 and #136

The reason this failed on CRAN now is that [`&&` and `||` now give a warning when one of the comparisons has a length greater than one](https://www.jumpingrivers.com/blog/new-features-r420/#changes-to-logical-functions)

**Comment by owenjonesuob** [comment]
Hi folks :wave: afaict this is the current situation:
* The version of {goodpractice} which was on CRAN prior to its removal was v1.0.2, released 2018-05-02 according to the [release archive](https://cran.r-project.org/src/contrib/Archive/goodpractice/) (this isn't the same as the GitHub release, which was rather posthumous)
* Due to changes to `&&`/`||`, in R-devel, some of CRAN's routine checks on v1.0.2 began failing earlier this year
* The commit which (super-preemptively!) fixed these issues was 09dd119 (@dpprdan you are absolutely correct)
* That commit was merged on 2019-11-14, i.e. long after v1.0.2 was released on CRAN, but long before the issues surfaced in R-devel checks
* There hasn't been a release of {goodpractice} since, so the fix never made it to CRAN
* We haven't yet successfully submitted this release (v1.0.3) to CRAN: we tried a month or two ago, but it got lost in the mail, according to dialogue on #154; and we haven't retried, because we were distracted by the (non-)problem I've just described

So I _think_ the very simple solution is that we just need to submit `v1.0.3` to CRAN (and to make sure it gets there this time :innocent:).

@KarinaMarks and/or @anasimmons (apologies, not sure who has ended up taking over maintenance), would you be able to do so at your earliest convenience?

---

FWIW this afternoon I installed R-devel on a Fedora machine I had lying around, and tried to build {goodpractice} just in case something else was wrong - but everything looks fine to me locally, so we'll hope it's the same on CRAN's end :+1:

**Comment by KarinaMarks** [comment]
Package has been uploaded to CRAN with latest updates - will merge once accepted. Thanks all for your hard work into getting this package back onto CRAN!

**Pull request merged**